src/luarocks/cmd/make.lua

```diff
@@ -84,8 +87,12 @@ function make.command(flags, rockspec_filename)
       branch = not not flags["branch"],
    })
 
+   if flags["sign"] and not flags["pack-binary-rock"] then
+      return nil, "In the make command, --sign is meant to be used only with --pack-binary-rock"
```

Comment:
why not allow people to sign a src rock?

Reply:
that's allowed, but not in the `make` command. You sign a src.rock with `luarocks pack --sign <rockspec>` (or via `luarocks upload --sign`)